python/grass/jupyter/interact_display.py

```diff
@@ -0,0 +1,228 @@
+        self._extent = self._convert_extent(env=os.environ) # Get the extent of the original area in WGS84
+        
+        # Set region to match original region extent
+        gs.run_command("g.region",
+                       n=self._extent["North"],
+                       s=self._extent["South"],
+                       e=self._extent["East"],
+                       w=self._extent["West"],
+                       env=self._env
+                      )
+        
+        # Get Center of tmp GRASS region
+        center = gs.parse_command("g.region", flags="cg", env=self._env)
+        center = (float(center["center_northing"]), float(center["center_easting"]))
+
+        # Create Folium Map 
+        self.map = folium.Map(
+            width=self.width,
+            height=self.height,
+            location=center,
+            tiles="cartodbpositron",
+        )
+
+        
+    def _get_rc(self, env=None):
+        rc = {}
+        
+        if env:
+            rcfile_location = env["GISRC"]
+        else:
+            rcfile_location = os.environ["GISRC"]
+            
+        with open(rcfile_location) as f:
+            for line in f:
+                entry = line.strip("\n").split(": ")
+                rc[entry[0]] = entry[1]
+        return rc
+    
+    
+
+    def _convert_coordinates(self, coordinates, proj_in):
+        """This function reprojects coordinates to WGS84, the required
+        projection for folium.
+
+        Coordinates should be a string with x and y values separated
+        by a comma.
+
+        proj_in is a proj4 string, for example, the output of g.region
+        with the `g` flag."""
+
+        coords_folium = gs.read_command(
+            "m.proj",
+            coordinates=coordinates,
+            proj_in=proj_in,
+            separator="comma",
+            flags="do",
+        )
+
+        # Reformat from string to array
+        coords_folium = coords_folium.strip()  # Remove '\n' at end of string
+        coords_folium = coords_folium.split(",")  # Split on comma
+        coords_folium = [float(value) for value in coords_folium]  # Convert to floats
+
+        return coords_folium[1], coords_folium[0]  # Return Lat and Lon
+
+    def _convert_extent(self, env=None):
+        """This function returns the bounding box of the current region
+        in WGS84, the required projection for folium"""
+
+        # Get proj of current GRASS region
+        proj = gs.read_command("g.proj", flags="jf", env=env)
+
+        # Get extent
+        extent = gs.parse_command("g.region", flags="g", env=env)
+        extent_ne = "{}, {}".format(extent["e"], extent["n"])
+        extent_sw = "{}, {}".format(extent["w"], extent["s"])
+
+        # Convert extent to EPSG:3857, required projection for Folium
+        north, east = self._convert_coordinates(extent_ne, proj)
+        south, west = self._convert_coordinates(extent_sw, proj)
+
+        extent = {'North': north, 'South': south, 'East': east, 'West': west}
```

Comment:
The upper case here is somewhat unexpected. Either all lower case or switch the dictionary to types.SimpleNamespace and use attributes where it is clear it needs to be lower case.